tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1060 @@
+       "        output: Tensor(torch.Size([5, 10]), dtype=torch.float32)},\n",
+       "    batch_size=torch.Size([5]),\n",
+       "    device=cpu,\n",
+       "    is_shared=False)"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tensordict = TensorDict(\n",
+    "    {\n",
+    "        \"a\": torch.randn(5, 3),\n",
+    "        \"b\": torch.randn(5, 4),\n",
+    "    },\n",
+    "    batch_size=[5],\n",
+    ")\n",
+    "\n",
+    "mergelinear = TensorDictModule(\n",
+    "    MergeLinear(3, 4, 10), in_keys=[\"a\", \"b\"], out_keys=[\"output\"]\n",
+    ")\n",
+    "\n",
+    "mergelinear(tensordict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11256ae7",
+   "metadata": {},
+   "source": [
+    "### Example 3: Multiple outputs\n",
+    "Similarly, `TensorDictModule` not only supports multiple inputs but also multiple outputs. To make a `TensorDictModule` instance write to multiple output values values, one must register them in the `out_keys` keyword argument of the constructor.\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "0b7f709b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MultiHeadLinear(nn.Module):\n",
+    "    def __init__(self, in_1, out_1, out_2):\n",
+    "        super().__init__()\n",
+    "        self.linear_1 = nn.Linear(in_1, out_1)\n",
+    "        self.linear_2 = nn.Linear(in_1, out_2)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        return self.linear_1(x), self.linear_2(x)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "1b2b465f",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "TensorDict(\n",
+       "    fields={\n",
+       "        a: Tensor(torch.Size([5, 3]), dtype=torch.float32),\n",
+       "        output_1: Tensor(torch.Size([5, 4]), dtype=torch.float32),\n",
+       "        output_2: Tensor(torch.Size([5, 10]), dtype=torch.float32)},\n",
+       "    batch_size=torch.Size([5]),\n",
+       "    device=cpu,\n",
+       "    is_shared=False)"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tensordict = TensorDict({\"a\": torch.randn(5, 3)}, batch_size=[5])\n",
+    "\n",
+    "splitlinear = TensorDictModule(\n",
+    "    MultiHeadLinear(3, 4, 10),\n",
+    "    in_keys=[\"a\"],\n",
+    "    out_keys=[\"output_1\", \"output_2\"],\n",
+    ")\n",
+    "splitlinear(tensordict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "859630c3",
+   "metadata": {},
+   "source": [
+    "As we shown previously, the `TensorDictModule` can take any `nn.Module` and perform the operations on a `TensorDict`. When having multiple input keys and output keys, make sure they match the order in the module.\n",
```

Comment:
not "the `TensorDictModule`" but "the `TensorDictModule` class" or "`TensorDictModule`"